core/src/interface/IExtensionTypes.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+interface IExtensionTypes {
+    /*//////////////////////////////////////////////////////////////
+                            STRUCTS & ENUMS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Enum for the type of call to be made to an extension function.
+    enum CallType {
+        STATICCALL,
+        CALL,
+        DELEGATECALL
+    }
+
+    /// @dev Struct for an extension function. Installing an extension in a core adds its extension functions to the core's ABI.
+    struct ExtensionFunction {
+        bytes4 fnSelector;
+        string fnSignature;
```

Comment:
Why do we need to store the fnSignature?

Reply:
We don't _need_ to -- one can always query https://www.4byte.directory/ with a selector value and get the function signature in return.

Is this something we would rely on in our stack? i.e. a call to 4byte directory? I've kept function signature as a struct field in the case that we don't want to rely on querying a database when building the joint ABI of a modular contract.